### ISSUE TITLE
what is the plan for gnome 43

### ISSUE THREAD

**Issue by RyzenDew** [kgshank/gse-sound-output-device-chooser#258]
![image](https://user-images.githubusercontent.com/84041391/188762188-63434184-062e-4862-b4b4-da432ecb8866.png)
as you can see they really made gnome 43 menu ugly and i am hoping you guys don't give up and fix this mess

**Comment by kgshank** [comment]
Tried creating a Poll, but couldn't get it work. So react to the following comments and I can understand your expectations

**Comment by kgshank** [comment]
React to this comment, if you want me to 
**Continue maintaining the extension as is for Gnome 43**
(Might me delayed as new gnome shell code is completely reworked)

**Comment by kgshank** [comment]
React to this comment, if you want me to
**Augment Gnome 43 with additional features**
(Comment on your desired features, so it will be easier for porting)

**Comment by kgshank** [comment]
React to this comment, if you want me to
**Extension not needed for Gnome 43**
(Will be maintained for Gnome < 42)

**Comment by RyzenDew** [comment]
Weird we can’t react ..

**Comment by kgshank** [comment]
> Weird we can’t react ..

I am not sure whats wrong, can you add an +1 comment with the option you need

**Comment by yosef406** [comment]
one feature that I like to see for current release and additional releases is:
**the ability to check or uncheck input output devices to stop them from showing**
it can be from the settings menu in the extension manager, it shouldn't disable the device.
I just installed the **noisetorch** software and it adds a few devices to the sound output (null out),
i want to use this extension to choose and switch between devices but i dont want the null to show up,
additionally i dont want all my devices to show (i have screens that have speakers that i never use),
the devices show in the extension settings it will be great if a check box was added to control this functionality.
thank you.

_i will add a comment for the pull_

**Comment by yosef406** [comment]
feature request:
**The ability to check or uncheck input output devices to stop them from showing**

**Comment by frandieguez** [comment]
Augment Gnome 43 with additional features

- Basically being able to change which elements are shown in the menu (show audio profiles, show/hide icons, hide certain output devices)

**Comment by thesmallcreeper** [comment]
So from what I see Gnome 43 allows functionality familiar to "Integrate selector with slider", haven't tested it yet.
https://www.omglinux.com/gnome-43-makes-it-much-easier-to-switch-audio-device/

In my opinion, since Gnome is willing to offer such functionality, additional features should be added to Gnome directly (not via extension). However that can be difficult because Gnome maintainers might don't want to get user experience bloated. So 
augment Gnome 43 with additional features might be the only option.

**Comment by RyzenDew** [comment]
gnome 43 did a horrible ugly job nothing changes that it's like it was made by a child and doesn't look good at all so i hope you fix it and de-ugly it

**Comment by YangtseSu** [comment]
Add the selection of audio inputs.

**Comment by razzeee** [comment]
I would really like to see current input output levels on each device, as that would make selecting the correct one easier. And it might also lead to the gnome devs being able to see if that's something they want to expand on.

**Comment by hendersj** [comment]
I would agree that audio input/output levels per device would be nice.  Also being able to rename the devices with a descriptive name (I don't know which monitor my "HDMI / DisplayPort 3" device is, but if I can name it "Dell Widescreen Monitor" once and forget about it, that would make life so much easier).

**Comment by razzeee** [comment]
The gnome implementation has input devices. But they are only showing, if it's active, so you have to have an app that's actually accessing the input, for it to show up in quick settings.

**Comment by hendersj** [comment]
That's kinda ugly, though, if you're wanting to select your input device before you start, say, a Zoom meeting....

**Comment by razzeee** [comment]
Feel free to voice your oppinion here then https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/5913
I guess we'll can get it to open up again if enought people feel like that.

**Comment by hendersj** [comment]
I'd be fine if this extension did it instead.  Doesn't have to be part of the core GNOME functionality. :)

**Comment by skgsergio** [comment]
I've been testing GNOME 43 and, design aside, what I mostly miss is the Input/Mic selector even when it is not in use.

I use this since my Headset has two profiles (Headphone and Handsfree) and for example, Microsoft Teams Web if the input change is not done beforehand when a call kicks in it doesn't list the Handsfree input device (yes, it is probably MS Teams fault, but I bypass this by changing first the input to the Handsfree).

**Comment by jim-moe** [comment]
I have found your extension to be a reliable way to change the sound in-/output. Using the Gnome settings for sound rarely actually changes the setting; quite annoying. With the advent of Gnome 43, I have no way to change the sound settings.

**Comment by sezcl** [comment]
+1 gnome 43 support.

**Comment by kpfleming** [comment]
The most recent release of GNOME Shell for GNOME 43 does have the ability to choose an output device from the Quick Settings widget (whether it works for @jim-moe or not is a different question!). It does not have the ability to do so for the input device unless an application is actively capturing input, which was the primary reason why I used this extension in the first place.

If we can convince the GNOME Shell team to give us a way to have the input device selector *always* visible in Quick Settings then I'd be satisfied with that and would not need an upgraded version of this Extension. It's sad to say that as it has served me very well for man years...

**Comment by jim-moe** [comment]
Hmm @kpfleming. I presume you mean the settings menu at the top left (of my screen)? If so, there is no option for selecting an in- or output device.

**Comment by kpfleming** [comment]
![image](https://user-images.githubusercontent.com/305938/196502739-fcc9f2e2-a389-4a28-9b24-40799aa71e76.png)
This is what the new Quick Settings widget looks like on my system with GNOME 43; the arrow to the right of the volume slider for the output is what I can click to select an output device.
![image](https://user-images.githubusercontent.com/305938/196502982-4e3be02f-29a6-4cd9-bc7d-8e860c6415fd.png)

**Comment by jim-moe** [comment]
Well. I fell silly. So it is here as well.

**Comment by kpfleming** [comment]
No worries, it's not as obvious as it could be and the GNOME team has some more work to do there. It's still not as good as the extension is, for example right now I have two 'Analog Output' devices listed in the Quick Settings panel with no way to distinguish them (not even a tooltip when hovering over them).

**Comment by muttleyxd** [comment]
One thing that is missing in Gnome 43 is any kind of hint, which device is the 'host' of given audio output/input.

I have several different audio output devices (microphone with loopback, HDMI/DisplayPort, integrated sound card, external DAC) and I can't tell which "Digital Output (S/PDIF)" is the one, that I want.

![Gnome 43 audio settings](https://i.imgur.com/qsKYDDt.png)

This extension was doing that just fine on Gnome 42, so I think that's one place where it could extend Gnome 43.

**Comment by darkdragon-001** [comment]
> One thing that is missing in Gnome 43 is any kind of hint, which device is the 'host' of given audio output/input.
> 
> I have several different audio output devices (microphone with loopback, HDMI/DisplayPort, integrated sound card, external DAC) and I can't tell which "Digital Output (S/PDIF)" is the one, that I want.
> 
> ![Gnome 43 audio settings](https://camo.githubusercontent.com/11e64117ede655fc6afd80b9a3137a4446d09d7cd2d43da6c1c9e8b41c79fd5b/68747470733a2f2f692e696d6775722e636f6d2f71734b594444742e706e67)
> 
> This extension was doing that just fine on Gnome 42, so I think that's one place where it could extend Gnome 43.

@muttleyxd This looks like something which should be filed as a bug upstream as well. Of course this shouldn't stop this extension from fixing this until an upstream patch is ready.

**Comment by skgsergio** [comment]
Regarding no distinction between outputs/inputs there is a bug filled upstream and a maintainer said that it will be fixed in 43.1: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/5829

**Comment by skgsergio** [comment]
As expected the disambiguation of inputs/outputs landed at 43.1:

![image](https://user-images.githubusercontent.com/186349/200179468-360d55c5-4607-46b3-82f3-eda2b7ff72ff.png)

**Comment by lviggiani** [comment]
> As expected the disambiguation of inputs/outputs landed at 43.1:
> 
> ![image](https://user-images.githubusercontent.com/186349/200179468-360d55c5-4607-46b3-82f3-eda2b7ff72ff.png)

Hi I just wanted to try it on gnome 43.1 (I've got it on Arch). I see that there's a branch named `43` but it seems identical to `master` also `metadata.json` doesn't list `43` among the compatible versions. So the question is, can I manually add `43` to `metadatra.json` in `master` or branch `43` or is there anything else I have to do? Thanks!

**Comment by razzeee** [comment]
@lviggiani that has been fixed in gnome itself and not in this extension

**Comment by lviggiani** [comment]
Sure, I know but I just wanted to test the extension. I tried forcing value
43 in metadata.json but it gives me an error when enabling it

Il mer 9 nov 2022, 16:57 Kolja Lampe ***@***.***> ha scritto:

**Comment by lviggiani** [comment]
> @lviggiani The easiest way to test would be to set the key `/org/gnome/shell/disable-extension-version-validation` to false using either the dconf-editor app or the dconf CLI tool. Then Gnome will allow you to load extensions who don't list the current shell version as compatible in their `metadata.json` file.

Sorry maybe I said that badly. I managed "starting" the extesion by adding the entry to metadata.json, but it gives this error:
`this._aggregateMenu is undefined` that I think has to do wtih GS 43 compatibility

**Comment by feimosi** [comment]
It'd be great to have this extension working because the new Gnome 43 menu doesn't allow switching the microphone source :/

**Comment by razzeee** [comment]
@feimosi *when your not using the microphone

**Comment by skgsergio** [comment]
@razzeee yes, but for some use cases is important to have it when not using it.

For example in MS Teams Web if I change my Bluetooth headset from the headphones profile (high audio quality, no mic) to headset (audio+mic) once it started capturing the mic I have to re join the meeting to start using the headset mic instead the laptop mic.

I used to change the audio output and input before joining a meeting with this extension.

For me, permanent input selection is the only thing I miss in vanilla GNOME 43.

**Comment by razzeee** [comment]
@skgsergio we already talked about that above and I think someone also raised the issue on the gnome tracker.

**Comment by RyzenDew** [comment]
Gnome never gives us what we need but this extension can plus even if they did it would never be done right

**Comment by jwest75674** [comment]
Additional Feature Request:
Would love to prioritize input and output devices, such that both are decided upon automatically.

**Comment by lig** [comment]
@jwest75674 that would be really neat feature to have. I've ended up blacklisting some devices in pulseaudio and configuring fallbacks manually to achieve what I need.

**Comment by jrom99** [comment]
Another thing that's really nice about this extension is the ability to select different profiles within a device, as this enables a soft profile reset, as well as the ability to hide useless devices and profiles.

My laptop has a nasty bluetooth bug where it changes my output device to a headset, but since GNOME doesn't recognize this change, the default A2DP profile is still selected. With the extension I was able to easily change profiles within A2DP and it'd go back to normal. Bonus points that I could also hide the headset device option, since my headphone doesn't even have a microphone.

**Comment by pablorq** [comment]
Just move the extension outside the new gnome panel and make it accessible directly. It also make the extension free from future changes on gnome.
I could be a new icon on the top bar, like pulse-audio-shortcuts extension.

**Comment by QmwJlHuSg9pa** [comment]
> Just move the extension outside the new gnome panel and make it accessible directly. It also make the extension free from future changes on gnome.

Agreed. The Quick Settings menu is overloaded as it is (in both functionality and visual noise). Given that this extension is geared towards "power users", a bespoke, customisable menu would be ideal--in terms of ease-of-use and muscle memory.

It would also be nice to have options for the following:

- Set location of volume sliders (top or bottom),
- Expand device lists by default (either output, input, or both)
- As others have mentioned, profile selection directly from the device lists

**Comment by kgshank** [comment]
Thanks for all the suggestions. Making it as a separate indicator makes more sense going forward, than depending on the Gnome implementation for volume sliders and device selection. I wanted to push some update before November, but couldn't spend much time with the extension. Unfortunately, I have other plans till end of January, so the updates will be further delayed. I hope to push something before the next Gnome release at least.

**Comment by iadegesso** [comment]
The one in GNOME 43 is buggy,, please port your extension to the newest GNOME

**Comment by esa1975** [comment]
My vote is to augment the existing layout in 43. I don't mind the direction they are taking but I do miss being able to easily choose audio devices. Thanks for asking for feedback from your users. Happy new year to you!

**Comment by 7system7** [comment]
For me, the GNOME 43's solution has been worked until the first minor version. Since then, I have option to change the sound output for one time. 🤣 After that I have 2 options
- I satisfied my chosen option forever
- Reboot my computer or restart the shell and change again

[Screencast from 2023-02-11 12-16-49.webm](https://user-images.githubusercontent.com/3964568/218255213-65749227-3a85-446e-bf73-7ac741cb4cea.webm)

Jokes aside; it is so boring that, in 2023 the GNOME team cannot implement anything without a massive bug. So.. I created a shortcut for the sound options.

**Comment by kgshank** [comment]
I had to take a longer break from the development than I expected, due to both personal and professional reasons.  The sad part that code that I was working for v43 was lost, as I had to format my machine. Now I am starting again from scratch. So I might have to skip 43 already completely and  probably release for 44 directly. I hope there are no delays further. Thanks everyone for your support and participation

**Comment by brianjmurrell** [comment]
Hopefully preaching to the choir here but also hopefully the lesson learned is to create a branch on GH for your development and push to it frequently (i.e. at least daily).  In that case, even with a re-format of your machine, your work would not have been lost.

**Comment by rbarte** [comment]
Fedora 38 runs on GNOME 44.1. When will GNOME 44 be supported?
